### PR TITLE
Quill Component System/ Site Banners

### PR DESCRIPTION
**Pull request by eadams17**
## WHAT
add site banners to the Quill Component Library

## WHY
we want to have a shared, reusable component for the various banners around our site

## HOW
add a new generic `Banner` component and add examples to Backpack

### Screenshots
(If applicable. Also, please censor any sensitive data)
Screen Shot 2023-11-15 at 5 46 42 PM
Screen Shot 2023-11-15 at 5 47 07 PM
Screen Shot 2023-11-15 at 5 47 26 PM
Screen Shot 2023-11-15 at 5 47 40 PM

### Notion Card Links
(Please provide links to any relevant Notion card(s) relevant to this PR.)
https://www.notion.so/quill/Quill-Component-System-Site-Banners-Update-to-Be-Coherent-with-the-new-Global-Header-829e85a390464dce8dad65b0264c5701

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? | yes
Have you deployed to Staging? | yes
Self-Review: Have you done an initial self-review of the code below on Github? | yes
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | yes

### CODE REVIEW

**Review by brendanshean — APPROVED** [review]
Looks good to me.  Very easy to follow.

File: services/QuillLMS/client/app/bundles/Shared/components/shared/banner.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        <p className="body">{bodyText}</p>
+        <div className="buttons-container">
+          {buttons.map((button, i) => {
+            const { onClick, href, standardButtonStyle, text, target } = button
+            const buttonClass = standardButtonStyle ? "quill-button fun primary contained" : "nonstandard-banner-button"
+            if(button.onClick) {
+              return <button className={`${buttonClass} focus-on-light`} key={`button-${i}`} onClick={onClick}>{text}</button>
+            } else {
+              return <a className={`${buttonClass} focus-on-light`} href={href} key={`button-${i}`} rel="noopener noreferrer" target={target}>{text} {!standardButtonStyle && <img alt={arrowPointingRightIcon.alt} src={arrowPointingRightIcon.src} />}</a>
+            }
```
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Shared/components/shared/banner.tsx]: I would pull this into a const above just for readability -- it feels like a lot for inline logic.

**Review by emilia-friedberg — COMMENTED** [review]
These look great overall! I left one small comment, and I have a more general question about naming. This might be something to discuss with Jack, but when I read "banner", I think about the elements that go above the navigation and extend across the whole page (like the `WebinarBanner`, `DemoAccountBanner`, etc). I think we should try to come up with a different name for this component. I was thinking something like "InPageBanner", but ChatGPT had more creative suggestions:

Screenshot 2023-11-16 at 9 06 10 AM

**Comment by eadams17** [comment]
> These look great overall! I left one small comment, and I have a more general question about naming. This might be something to discuss with Jack, but when I read "banner", I think about the elements that go above the navigation and extend across the whole page (like the `WebinarBanner`, `DemoAccountBanner`, etc). I think we should try to come up with a different name for this component. I was thinking something like "InPageBanner", but ChatGPT had more creative suggestions:
> 
> Screenshot 2023-11-16 at 9 06 10 AM

So I discussed with Jack, and I think we still want to call this a banner since it stretches across the whole page. My idea was to have `PreNavigationBanner` used for things like webinars and the demo account, and this component could be the `PostNavigationBanner`. Thoughts?

**Review by emilia-friedberg — APPROVED** [review]

**Pull request merged**